contracts/auctions/src/storage.rs

```diff
@@ -0,0 +1,290 @@
+    let limit = limit.unwrap_or(DEFAULT_LIMIT);
+
+    let mut auctions = vec![&env];
+
+    for id in start_index..=limit {
+        match get_auction_by_id(env, id as u64) {
+            Ok(auction) => auctions.push_back(auction),
+            Err(ContractError::AuctionNotFound) => break,
+            Err(e) => return Err(e),
+        }
+    }
+
+    Ok(auctions)
+}
+
+pub fn save_auction_by_id(
+    env: &Env,
+    auction_id: u64,
+    auction: &Auction,
+) -> Result<(), ContractError> {
+    env.storage().instance().set(&auction_id, auction);
+    env.storage()
+        .instance()
+        .extend_ttl(LIFETIME_THRESHOLD, BUMP_AMOUNT);
+
+    Ok(())
+}
+
+pub fn save_auction_by_seller(
+    env: &Env,
+    seller: &Address,
+    auction: &Auction,
+) -> Result<(), ContractError> {
+    let mut seller_auctions_list: Vec<Auction> =
+        env.storage().instance().get(seller).unwrap_or(vec![&env]);
+
+    match seller_auctions_list.iter().position(|a| a.id == auction.id) {
+        Some(existing_idx) => seller_auctions_list.set(existing_idx as u32, auction.clone()),
+        None => seller_auctions_list.push_back(auction.clone()),
+    };
+
+    env.storage().instance().set(seller, &seller_auctions_list);
+
+    env.storage()
+        .instance()
+        .extend_ttl(LIFETIME_THRESHOLD, BUMP_AMOUNT);
+
+    Ok(())
+}
+
+pub fn get_auction_by_id(env: &Env, auction_id: u64) -> Result<Auction, ContractError> {
+    let auction = env
+        .storage()
+        .instance()
+        .get(&auction_id)
+        .unwrap_or_else(|| {
+            log!(env, "Auction: Get auction by id: Auction not present");
+            panic_with_error!(&env, ContractError::AuctionNotFound);
+        });
+    env.storage()
+        .instance()
+        .extend_ttl(LIFETIME_THRESHOLD, BUMP_AMOUNT);
+
+    auction
+}
+
+pub fn get_auctions_by_seller_id(
+    env: &Env,
+    seller: &Address,
+) -> Result<Vec<Auction>, ContractError> {
+    let seller_auctions_list = env.storage().instance().get(seller).unwrap_or_else(|| {
+        log!(env, "Auction: Get auction by seller: No auctions found");
+        panic_with_error!(&env, ContractError::AuctionNotFound);
+    });
+    env.storage()
+        .instance()
+        .extend_ttl(LIFETIME_THRESHOLD, BUMP_AMOUNT);
+
+    Ok(seller_auctions_list)
+}
+
+pub fn update_auction(env: &Env, id: u64, auction: Auction) -> Result<(), ContractError> {
+    if id != auction.id {
+        log!(&env, "Auction update auction: Id missmatch");
+        panic_with_error!(&env, ContractError::IDMissmatch);
+    }
+    env.storage().instance().set(&auction.id, &auction);
+    env.storage()
+        .instance()
+        .extend_ttl(LIFETIME_THRESHOLD, BUMP_AMOUNT);
+
+    Ok(())
+}
+
+pub fn validate_input_params(env: &Env, values_to_check: &[&u64]) -> Result<(), ContractError> {
+    values_to_check.iter().for_each(|i| {
+        if i < &&1 {
+            log!(&env, "Auction: Validate input: Invalid inputs used");
```

Comment:
Since you are checking the specific case, you don't need to be so general.
```suggestion
            log!(&env, "Auction: Validate input: parameters can not be less then 1");
```